File: shared/src/validators.ts
const YOUTUBE_PATTERNS = [
  /youtube\.com\/watch\?.*v=/,
  /youtu\.be\//,
  /youtube\.com\/shorts\//,
  /youtube\.com\/playlist\?.*list=/,
];

export function isValidYouTubeUrl(url: string): boolean {
  return YOUTUBE_PATTERNS.some((p) => p.test(url));
}

export function isPlaylistUrl(url: string): boolean {
  return /youtube\.com\/playlist\?.*list=/.test(url);
}

File: legacy/utils/validators.py
import re

_YOUTUBE_PATTERNS = [
    r"youtube\.com/watch\?.*v=",
    r"youtu\.be/",
    r"youtube\.com/shorts/",
    r"youtube\.com/playlist\?.*list=",
]


def is_valid_youtube_url(url: str) -> bool:
    return any(re.search(p, url) for p in _YOUTUBE_PATTERNS)
